### ISSUE TITLE
Grappler error when Softmax is used in a custom layer

### ISSUE THREAD

**Issue by JustASquid** [tensorflow/tensorflow#57975]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Windows 10

### Python version

3.9

### CUDA/cuDNN version

11.6

### Current Behaviour?

```shell
Error/Warning from Grappler: `Error in PredictCost() for the op: op: "Softmax" attr { key: "T" value { type: DT_FLOAT } } inputs { dtype: DT_FLOAT shape { unknown_rank: true } } device { type: "GPU" vendor: "NVIDIA" model: "NVIDIA GeForce GTX 1080 Ti" frequency: 1670 num_cores: 28 environment { key: "architecture" value: "6.1" } environment { key: "cuda" value: "11020" } environment { key: "cudnn" value: "8100" } num_registers: 65536 l1_cache_size: 24576 l2_cache_size: 2883584 shared_memory_size_per_multiprocessor: 98304 memory_size: 10099490816 bandwidth: 484440000 } outputs { dtype: DT_FLOAT shape { unknown_rank: true } }`
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np

class Chooser(tf.keras.layers.Layer):
    @tf.function
    def call(self, options_input, choices_input_logits):
        choices = tf.nn.softmax(choices_input_logits, axis=2)

        result = tf.linalg.matmul(choices, options_input)
        return result

def get_model():
    options_input = tf.keras.layers.Input(shape=(10,3), name="options")
    choices_input = tf.keras.layers.Input(shape=(5,10), name="choices")

    net = Chooser()(options_input, choices_input)
    net = tf.keras.layers.Flatten()(net)
    net = tf.keras.layers.Dense(1)(net)

    return tf.keras.Model([options_input, choices_input], net)

model = get_model()
model.compile(
    optimizer=tf.keras.optimizers.Adam(0.001),
    loss=tf.keras.losses.MeanAbsoluteError(),
)

def batch_gen():
    while True:
        o = np.random.uniform(low=-1.0, high=1.0, size=(10, 3))
        c = np.random.uniform(low=0.0, high=1.0, size=(5, 10))

        y = 1

        yield {"options": o, "choices": c}, y

dataset = tf.data.Dataset.from_generator(batch_gen, output_types=({"options": tf.float32, "choices": tf.float32}, tf.float32))
dataset = dataset.batch(32)

model.fit(dataset, steps_per_epoch=100, epochs=1)
```
```


### Relevant log output

```shell
`Error in PredictCost() for the op: op: "Softmax" attr { key: "T" value { type: DT_FLOAT } } inputs { dtype: DT_FLOAT shape { unknown_rank: true } } device { type: "GPU" vendor: "NVIDIA" model: "NVIDIA GeForce GTX 1080 Ti" frequency: 1670 num_cores: 28 environment { key: "architecture" value: "6.1" } environment { key: "cuda" value: "11020" } environment { key: "cudnn" value: "8100" } num_registers: 65536 l1_cache_size: 24576 l2_cache_size: 2883584 shared_memory_size_per_multiprocessor: 98304 memory_size: 10099490816 bandwidth: 484440000 } outputs { dtype: DT_FLOAT shape { unknown_rank: true } }`
```
</details>

**Comment by JustASquid** [comment]
Hey @tilakrayal the platform is Windows 10 so I won't be able to run that command.

**Comment by tilakrayal** [comment]
@JustASquid,
Could you please create a virtual environment and test your code again. The attached [gist](https://colab.research.google.com/gist/tilakrayal/b0dda043a9e94eaeec91277966805d96/untitled646.ipynb) executed without any error, then there is no issue with the code and the Tensorflow. Thank you!

**Comment by JustASquid** [comment]
@tilakrayal - I have run this in a fresh virtual environment following the recommended GPU installation on Windows Native:
https://www.tensorflow.org/install/pip#windows-native_1

And the issue still comes up. It seems like this is a Windows issue, so it makes sense that the colab gist works correctly.

**Comment by tilakrayal** [comment]
@JustASquid,
We can see you are trying to install tensorflow v2.10 with Cuda 11.6. Could you try to install with tested build configurations as mentioned [here](https://www.tensorflow.org/install/source_windows#gpu) and try to execute the code. Thank you!

**Comment by gpassero** [comment]
I am having the same problem. Running Tensorflow 2.15 with cudnn 8600. Full log:

> W tensorflow/core/grappler/costs/op_level_cost_estimator.cc:693] Error in PredictCost() for the op: op: "Softmax" attr { key: "T" value { type: DT_HALF } } inputs { dtype: DT_HALF shape { unknown_rank: true } } device { type: "GPU" vendor: "NVIDIA" model: "NVIDIA A40" frequency: 1740 num_cores: 84 environment { key: "architecture" value: "8.6" } environment { key: "cuda" value: "11080" } environment { key: "cudnn" value: "8600" } num_registers: 65536 l1_cache_size: 24576 l2_cache_size: 6291456 shared_memory_size_per_multiprocessor: 102400 memory_size: 45716668416 bandwidth: 696096000 } outputs { dtype: DT_HALF shape { unknown_rank: true } }
